tutorial/lesson_21_auto_scheduler_generate.cpp

```diff
@@ -95,31 +95,33 @@ class AutoScheduled : public Halide::Generator<AutoScheduled> {
             // schedule will be.
 
             // To auto-schedule the pipeline, we don't have to do anything else:
-            // every Generator implicitly has a GeneratorParam named "auto_schedule";
-            // if this is set to true, Halide will call auto_schedule() on all of
-            // our pipeline's outputs automatically.
-
-            // Every Generator also implicitly has a GeneratorParams named "machine_params",
-            // which allows you to specify characteristics of the machine architecture
-            // for the auto-scheduler; it's generally specified in your Makefile.
+            // every Generator implicitly has a GeneratorParam named "auto_scheduler.name";
+            // if this is set to the name of the Autoscheduler we want to use, Halide will
+            // apply it to all of our pipeline's outputs automatically.
+
+            // Every Generator also implicitly has additional, optional GeneratorParams that are
+            // dependent on the specific Autoscheduler select, which allows you to specify
+            // characteristics of the machine architecture
+            // for the autoscheduler; it's generally specified in your Makefile.
             // If none is specified, the default machine parameters for a generic CPU
-            // architecture will be used by the auto-scheduler.
+            // architecture will be used by the autoscheduler.
 
-            // Let's see some arbitrary but plausible values for the machine parameters.
+            // Let's see some arbitrary but plausible values for the machine parameters
+            // for the Mullapudi2016 Autoscheduler:
             //
-            //      const int kParallelism = 32;
-            //      const int kLastLevelCacheSize = 16 * 1024 * 1024;
-            //      const int kBalance = 40;
-            //      MachineParams machine_params(kParallelism, kLastLevelCacheSize, kBalance);
+            //      autoscheduler.name=Mullapudi2016
```

Comment:
Hmm, that's an interesting question. I went with the ".name" approach for consistency inside the code (ie when accessing the dict of autoscheduler-specific stuff), but... as a practical matter, I don't think that's ~ever likely to be visible to coders writing Generators. (I mean, you *could* inspect the name and, say, modify the estimates you provide based on the autoscheduler you're using, but it seems like a pretty fringy case.)

Let me go try it that way and see how it feels.

Reply:
OK, so I tried this and decided it was better... then decided to also rework AutoschedulerParams as a struct (with a name field) rather than just a plain string-to-string map (with name being a 'required' key). Not sure if better or not? PTAL.